### PR TITLE
Extract more meaningful modules from `model::cfd`

### PR DESCRIPTION
**Pull request by luckysori**
I set out to start tackling #1430, but I got carried away and started trying to remove things that didn't make sense any longer to me (such as the `Completed` enum) rather than just moving them around.

I intend to continue with #1430 in follow-up PRs. If reviewers already have some ideas for meaningful modules, that would be helpful, although I would action that feedback in the follow-up PRs.

### CODE REVIEW

File: model/src/hex_transaction.rs [hunk 1]
```diff
@@ -0,0 +1,50 @@
+use super::*;
+use bdk::bitcoin;
+use bdk::bitcoin::Transaction;
+use serde::Deserializer;
+use serde::Serializer;
+
```
Comment (thomaseizinger) [inline model/src/hex_transaction.rs:6]: We should usptream this.

File: daemon/src/setup_maker.rs [hunk 2]
```diff
@@ -117,24 +115,45 @@ impl Actor {
 
         self.tasks.add(async move {
             let _: Result<(), xtra::Disconnected> = match contract_future.await {
-                Ok(dlc) => this.send(SetupSucceeded { order_id, dlc }).await,
-                Err(error) => this.send(SetupFailed { order_id, error }).await,
+                Ok(dlc) => this.send(SetupSucceeded { dlc }).await,
+                Err(error) => this.send(SetupFailed { error }).await,
```
Comment (thomaseizinger) [inline daemon/src/setup_maker.rs:119]: Tiny nit: This could have been a separate commit.

**Review by thomaseizinger — APPROVED** [review]
Nice work!

**Review by klochowicz — APPROVED** [review]
great refactor! thanks for giving much needed TLC to the `model` <3

**Review by rishflab — APPROVED** [review]
Thanks, will be much easier to navigate the codebase !

File: model/src/hex_transaction.rs (hunk 1 above)
Comment (luckysori) [reply model/src/hex_transaction.rs:6]: Tracking this locally.

**Pull request merged**